sim/team-validator.ts

```diff
@@ -2693,7 +2693,9 @@ export class TeamValidator {
 		if (!setSources.restrictiveMoves) {
 			setSources.restrictiveMoves = [];
 		}
-		setSources.restrictiveMoves.push(move.name);
+		if (!setSources.restrictiveMoves.includes(move.name)) {
```

Comment:
Might want to change `restrictiveMoves` from an array to a Set. The lengths I've observed are 1 - 40, heavily weighted towards the low end. Linear scan on a small array isn't too bad, and sometimes better than a hashset. I haven't checked all the usages to see if switching to a Set is reasonable, and can't be sure that there isn't some scenario that grows `restrictiveMoves` to hundreds or thousands of elements.